lib/ace/layer/text.js

```diff
@@ -204,6 +204,8 @@ var Text = function(parentEl) {
                 );
                 lineElement.style.height = config.lineHeight * this.session.getRowLength(row) + "px";
                 lineElement.innerHTML = html.join("");
+                if (this.session.$bidiHandler.hasRtlSupport())
+                    this.setRtlDir({childNodes: [lineElement], textContent: lineElement.textContent});
```

Comment:
would be better to do this without accessing textContent, because getting the same information from line is faster